### PR TITLE
#1964

### PR DESCRIPTION
**Pull request by philbuettner**
Added Things resource: /things &  /things/{thingId}

### CODE REVIEW

File: generators/generator-openapi/src/main/java/org/eclipse/vorto/codegen/openapi/templates/OpenAPITemplate.xtend [hunk 1]
```diff
@@ -373,6 +482,30 @@ class OpenAPITemplate implements IFileTemplate<InformationModel> {
...
+		      - features
+		      properties:
+		        thingId:
+		          type: string
+		          description: Unique identifier representing the thing
+		        policyId:
+		          type: string
+		          description: The policy ID used for controlling access to this thing, managed by resource `/policies/{policyId}`
+		        attributes:
+		          $ref: '#/components/schemas/Attributes'
```
Comment (aedelmann) [inline generators/generator-openapi/src/main/java/org/eclipse/vorto/codegen/openapi/templates/OpenAPITemplate.xtend]: Where is attributes defined ?

File: generators/generator-openapi/src/main/java/org/eclipse/vorto/codegen/openapi/templates/OpenAPITemplate.xtend [hunk 2]
```diff
@@ -373,6 +482,30 @@ class OpenAPITemplate implements IFileTemplate<InformationModel> {
...
+		        features:
+		          $ref: '#/components/schemas/Features'
```
Comment (aedelmann) [inline generators/generator-openapi/src/main/java/org/eclipse/vorto/codegen/openapi/templates/OpenAPITemplate.xtend]: Where is Features defined ? Shouldn't this be the «infomodel.name»Features ?

Comment (philbuettner) [reply generators/generator-openapi/src/main/java/org/eclipse/vorto/codegen/openapi/templates/OpenAPITemplate.xtend]: Attributes is defined in line 503.

File: generators/generator-openapi/src/main/java/org/eclipse/vorto/codegen/openapi/templates/OpenAPITemplate.xtend (hunk 1 above)
Comment (philbuettner) [reply generators/generator-openapi/src/main/java/org/eclipse/vorto/codegen/openapi/templates/OpenAPITemplate.xtend]: Attributes is defined in line 503.

File: generators/generator-openapi/src/main/java/org/eclipse/vorto/codegen/openapi/templates/OpenAPITemplate.xtend (hunk 2 above)
Comment (philbuettner) [reply generators/generator-openapi/src/main/java/org/eclipse/vorto/codegen/openapi/templates/OpenAPITemplate.xtend]: I will add FeatureDefinition and FeatureProperties in a next step. to be in line with the IoT Things schema.

Comment (aedelmann) [reply generators/generator-openapi/src/main/java/org/eclipse/vorto/codegen/openapi/templates/OpenAPITemplate.xtend]: Hi @philbuettner 
AFAIK the Features specifically for the information model are already generated. In this case the Thing model would directly reference that one. The one you said from existing Things Schema is a generic structure , and not strong typed to the function blocks.

Comment (philbuettner) [reply generators/generator-openapi/src/main/java/org/eclipse/vorto/codegen/openapi/templates/OpenAPITemplate.xtend]: Ok, what should be the content of /things/{thingId} then?

Comment (aedelmann) [reply generators/generator-openapi/src/main/java/org/eclipse/vorto/codegen/openapi/templates/OpenAPITemplate.xtend]: the same Thing model as you have specified except that the **features** references #schema/components/«infomodel.name»Features

**Comment by aedelmann** [comment]
@philbuettner Some of your commits are not signed off. Can you do that, otherwise I cannot merge.

**Pull request closed without merging**